code/datums/elements/elevation.dm

```diff
@@ -0,0 +1,169 @@
+	if(!HAS_TRAIT(location, TRAIT_TURF_HAS_ELEVATED_OBJ(pixel_shift)))
+		UnregisterSignal(location, list(COMSIG_TURF_RESET_ELEVATION, COMSIG_TURF_CHANGE))
+		reset_elevation(location)
+
+///Changing or destroying the turf detaches the element, also we need to reapply the traits since they don't get passed down.
+/datum/element/elevation/proc/pre_change_turf(turf/changed, path, list/new_baseturfs, flags, list/post_change_callbacks)
+	SIGNAL_HANDLER
+	var/list/trait_sources = GET_TRAIT_SOURCES(changed, TRAIT_TURF_HAS_ELEVATED_OBJ(pixel_shift))
+	trait_sources = trait_sources.Copy()
+	post_change_callbacks += CALLBACK(src, PROC_REF(post_change_turf), trait_sources)
+
+/datum/element/elevation/proc/post_change_turf(list/trait_sources, turf/changed)
+	ADD_TRAIT(changed, TRAIT_TURF_HAS_ELEVATED_OBJ(pixel_shift), trait_sources)
+	reset_elevation(changed)
+
+#define ELEVATE_TIME 0.2 SECONDS
+
+/**
+ * The core element attached to the turf itself. Do not use this directly!
+ *
+ * Causes mobs walking over a turf with this element to be pixel shifted vertically by the pixel_shift amount.
+ * Because of the way it's structured, it should only be added through the elevation element (without the core suffix).
+ *
+ * To explain: in the case of multiple objects with (different instances of) the element being stacked on one turf somehow,
+ * we only want that with the highest pixel shift value to apply it to the turf, so that the mobs standing on top of it all
+ * doesn't look like it's floating off the pile.
+ */
+/datum/element/elevation_core
+	element_flags = ELEMENT_BESPOKE | ELEMENT_DETACH_ON_HOST_DESTROY
+	argument_hash_start_idx = 2
+	///The amount of pixel_z applied to the mob standing on the turf.
+	var/pixel_shift
+
+/datum/element/elevation_core/Attach(datum/target, pixel_shift)
+	. = ..()
+	if(!isturf(target))
+		return ELEMENT_INCOMPATIBLE
+	if(!pixel_shift)
+		CRASH("attempted attaching /datum/element/elevation_core with a pixel_shift value of [isnull(pixel_shift) ? "null" : 0]")
+
+	RegisterSignal(target, COMSIG_ATOM_ENTERED, PROC_REF(on_entered))
+	RegisterSignal(target, COMSIG_ATOM_AFTER_SUCCESSFUL_INITIALIZED_ON, PROC_REF(on_initialized_on))
+	RegisterSignal(target, COMSIG_ATOM_EXITED, PROC_REF(on_exited))
+	RegisterSignal(target, COMSIG_TURF_RESET_ELEVATION, PROC_REF(on_reset_elevation))
+
+	src.pixel_shift = pixel_shift
+
+	ADD_TRAIT(target, TRAIT_ELEVATED_TURF, REF(src))
+
+	for(var/mob/living/living in target)
+		RegisterSignal(living, COMSIG_LIVING_SET_BUCKLED, PROC_REF(on_set_buckled))
+		elevate_mob(living)
+
+/datum/element/elevation_core/Detach(datum/source)
+	/**
+	 * Since the element can be removed outside of Destroy(),
+	 * and even then, signals are passed down to the new turf,
+	 * it's necessary to clear them here.
+	 */
+	UnregisterSignal(source, list(
+		COMSIG_ATOM_ENTERED,
+		COMSIG_ATOM_AFTER_SUCCESSFUL_INITIALIZED_ON,
+		COMSIG_ATOM_EXITED,
+		COMSIG_TURF_RESET_ELEVATION,
```

Comment:
alphabetize these

Reply:
@Ghommie Bruh

Reply:
Sorry, it's been a bit of a busy week for me.